net/dns/config.go

```diff
@@ -91,6 +91,24 @@ func (c Config) hasDefaultIPResolversOnly() bool {
 	return true
 }
 
+func (c Config) hasHostsWithoutSplitDNSRoutes() bool {
```

Comment:
So in the case where no extra dns records are set, this returns false? (Which means we follow current expected paths.)

Reply:
yes. No change unless this returns true.